clients/go/commands/createJob_command.go

```diff
@@ -0,0 +1,144 @@
+package commands
+
+import (
+    "context"
+    "fmt"
+    "github.com/zeebe-io/zeebe/clients/go/pb"
+    "github.com/zeebe-io/zeebe/clients/go/utils"
+    "time"
+)
+
+type DispatchCreateJobCommand interface {
+    Send() (*pb.CreateJobResponse, error)
+}
+
+type CreateJobCommandStep1 interface {
+    JobType(string) CreateJobCommandStep2
+}
+
+type CreateJobCommandStep2 interface {
+    DispatchCreateJobCommand
+
+    SetCustomHeaderFromString(string) (CreateJobCommandStep2, error)
```

Comment:
These methods should be plural, i.e. `SetCustomHeaders*` as you override the whole map and not adding a single header.